plugins/tests/pod/test_pod.py

```diff
@@ -278,3 +278,46 @@ def simple_pod_task(i: int):
     assert serialized.template.k8s_pod.metadata.labels == {"label": "foo"}
     assert serialized.template.k8s_pod.metadata.annotations == {"anno": "bar"}
     assert serialized.template.k8s_pod.pod_spec is not None
+
+
+def test_map_pod_task_serialization():
```

Comment:
it doesn't make complete sense to add this test here as opposed to [test_map_task](https://github.com/flyteorg/flytekit/blob/master/tests/flytekit/unit/core/test_map_task.py) in core/ - but then that introduces a plugin dependency in core tests and I'm not sure what the lesser evil is.

Reply:
i completely get it, the thing is K8sPodSpec now is a core plugin.
Maybe the core task should support taking a yaml / json and the k8spod plugin can continue using the python interface to improve it. This way the flytekit remains devoid of k8s dependency

Reply:
hm not sure if i'm understanding correctly, but you mean have serialize() formulate the k8sPod target from a PodSpec struct? I'm not really a fan of making code more confusing (in this case, splitting up how we populate the k8sPod target across methods) just to make testing easier